### PR TITLE
[LTB-65] Add Buildable for ConfigRec

### PR DESCRIPTION
**Pull request by worm2fed**
Problem:
This was made here #53. But this PR became still.
It'll be good to proceed this PR and add Buildable instances, cause it's quite useful.
We're not going to merge #53, because Json instances was added recently independent from this PR (we didn't know about it).
    
Solution:
Move Buildable instances (along with tests) form #53.

Also in thins PR presented one stylistic commit with formatter config.

### CODE REVIEW

File: code/config/lib/Loot/Config/Buildable.hs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+instance OptionsToBuilderList (k :: ConfigKind) '[] where
+  configToBuilderList RNil = []
+
+undefinedBuilder :: Builder
+undefinedBuilder = build @Text "<undefined>"
+
+instance
+  forall l v is k
+  . ( KnownSymbol l
```
Comment (pasqu4le) [inline code/config/lib/Loot/Config/Buildable.hs:33]: One nitpick: here and in the instances that follow the explicit `forall` seems unnecessary, so I'd suggest to remove it.

File: code/config/lib/Loot/Config/Buildable.hs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    , Buildable v
+    , OptionsToBuilderList k is
+    )
+  => OptionsToBuilderList k ((l ::: v) ': is)
+  where
+    configToBuilderList (itemOption :& rest) =
+      let label = fromString $ symbolVal (Proxy :: Proxy l)
```
Comment (pasqu4le) [inline code/config/lib/Loot/Config/Buildable.hs:40]: One more suggestion: there are several uses of
```haskell
fromString $ symbolVal (Proxy :: Proxy l)
```
that could be extracted to a function.

However that function already exists, it's `getConfigKey` in `Loot.Config.Yaml`.

So I suggest moving `getConfigKey` into `Loot.Config.Record` and use it in this module and `Loot.Config.Yaml`.

**Review by pasqu4le — COMMENTED** [review]
Just a couple of small suggestions, but LGTM.

**Comment by pasqu4le** [comment]
Ah one more thing, I think that the commit message for b4e16fee42c9e3cd48f77705e119b263000df963 has a typo: 
_indention_ should be _indentation_.

Also, there's no written rule AFAIK, but in several projects for these commits that are not directly related to an issue we use the "tag" `[Chore]` at the beginning of the message instead, but I'll let you choose if use it here too or not.

File: code/config/lib/Loot/Config/Buildable.hs (hunk 2 above)
Comment (worm2fed) [reply code/config/lib/Loot/Config/Buildable.hs:40]: good!
I forgot about this function, thanks

**Comment by worm2fed** [comment]
> indention should be indentation.

https://www.merriam-webster.com/dictionary/indention

`indention` is an archaic form of `indentation`)

**Comment by worm2fed** [comment]
@pasqu4le if you want, you can take a look at requested changes

**Comment by pasqu4le** [comment]
> indention is an archaic form of indentation

oh I didn't know that, I assumed it was a typo. No problem then :ok_hand:

**Review by pasqu4le — APPROVED** [review]
LGTM now :+1:

**Pull request merged**